### PR TITLE
feat: add in-memory GraphQL server

### PR DESCRIPTION
**Pull request by xilosada**
## Summary:

This PR integrates an GraphQL server with Actix-web and async-graphql for handling of posts and comments.
For simplicity it stores everything in-memory.
It also includes a schema for GraphQL clients.

## Test Plan:

```
cargo run                                         
    Finished dev [unoptimized + debuginfo] target(s) in 0.46s
     Running `/Users/xabi/dev/cali2.0-experimental/target/debug/peer-space`
GraphiQL IDE: http://localhost:8000
```

Using the graphQL playground the following operations are available:

```
query {
  posts {
    id
    title
    content
    comments {
      user
      text
    }
  }
}
```
```
query {
  post(id: 0) {
    id
    title
    content
    comments {
      user
      text
    }
  }
}
```

```
mutation {
  createPost(input:{ title: "New Post", content: "Content of the new post"}) {
    id
    title
    content
  }
}

```

```
mutation {
  createComment(input: { post: 0, user: "Xabi", text: "FFF"}) {
    id
    title
    content
    comments {
      user
      text
    }
  }
}
```

Screenshot 2024-02-25 at 6 35 18 PM

### CODE REVIEW

File: peer-space/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+schema {
+    query: Query,
+    mutation: Mutation
+}
+
+type Query {
+    posts: [Post!]!
```
Comment (frdomovic) [inline peer-space/schema.graphql:8]: Can also "post" be added ? to get specific by id?

File: peer-space/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+input CreateComment {
+    post: ID!,
+    user: String,
+    text: String!
+}
+
+type Comment {
+    text: String,
```
Comment (MatejVukosav) [inline peer-space/schema.graphql]: comment should also have an id

File: peer-space/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+input CreatePost {
+    title: String!,
+    content: String!,
+}
+
+input CreateComment {
+    post: ID!,
+    user: String,
```
Comment (MatejVukosav) [inline peer-space/schema.graphql]: User cannot be null => String!

File: peer-space/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    text: String!
+}
+
+type Comment {
+    text: String,
+    user: String,
```
Comment (MatejVukosav) [inline peer-space/schema.graphql]: both text and user should be mandatory

File: peer-space/schema.graphql (hunk 1 above)
Comment (xilosada) [reply peer-space/schema.graphql:8]: added

File: peer-space/schema.graphql (hunk 2 above)
Comment (xilosada) [reply peer-space/schema.graphql]: we don't have any query or mutation that uses the ID. For posts we use it to add comments. Once we have a proper db and the ids can be consistant (right now is just the index in an array) we can add it if there is any use for it

**Pull request merged**